File: Codewars/Python/Range_sorting_string.py
def solution(args):
    result = ""
    range_in_array = []
    for i in range(len(args) - 1):
        if (abs(args[i] - args[i + 1]) == 1) and (i < len(args) - 1):
            range_in_array.append(args[i])
        else:
            range_in_array.append(args[i])

            if len(range_in_array) >= 3:
                result += str(range_in_array[0]) + '-' + str(range_in_array[len(range_in_array) - 1]) + ','
            
            else:
                for j in range_in_array:
                    result += str(j) + ','
            
            range_in_array.clear()

    range_in_array.append(args[-1])
    if len(range_in_array) >= 3:
        result += str(range_in_array[0]) + '-' + str(range_in_array[-1]) + ','
    else:
        for j in range_in_array:
            result += str(j) + ','


    return result[:-1:]

File: Codewars/CPP/Range_sorting_string.cpp
#include <string>
#include <vector>

std::string range_extraction(std::vector<int> args) {
	std::string result;
	std::vector <int> range;

	for (size_t i = 0; i < args.size(); i++)
	{
		if (i < args.size() - 1 && abs(args[i] - args[i + 1]) == 1)
		{
			range.push_back(args[i]);
		}
		else
		{
			range.push_back(args[i]);

			
			if (range.size() >= 3)
			{
				result += std::to_string(range[0]) + '-' + std::to_string(range[range.size() - 1]) + ",";
			}
			else
			{
				for (size_t j = 0; j < range.size(); j++)
				{
					result += std::to_string(range[j]) + ",";
				}
			}

			range.clear();
		}
	}

	result.pop_back();
	
	
	return result;
}